quantum-viz/quantum_viz/qiskit_parser.py

```diff
@@ -0,0 +1,370 @@
+import json
+from enum import IntEnum
+from typing import Dict
+from typing import Iterator
+from typing import List
+from typing import Optional
+from typing import Type
+from typing import Union
+
+try:
+    import qiskit  # noqa: F401
+except ImportError:
+    raise ImportError(
+        '`qiskit` was not found, try to `pip install "quantum-viz[qiskit]"`'
+    )
+
+from qiskit.circuit import QuantumCircuit, Qubit, Clbit, ClassicalRegister
+from qiskit.circuit.instruction import Instruction
+from qiskit.circuit.controlledgate import ControlledGate
+from qiskit.circuit.measure import Measure
+from qiskit.circuit.barrier import Barrier
+from qiskit.circuit.reset import Reset
+from qiskit.circuit.parameter import Parameter
+from qiskit.circuit.library import (
+    IGate,
+    XGate,
+    YGate,
+    ZGate,
+    HGate,
+    UGate,
+    U1Gate,
+    U2Gate,
+    U3Gate,
+    PhaseGate,
+    RGate,
+    SwapGate,
+    RXGate,
+    RYGate,
+    RZGate,
+    RXXGate,
+    RYYGate,
+    RZXGate,
+    RZZGate,
+    iSwapGate,
+    DCXGate,
+    SGate,
+    SdgGate,
+    TGate,
+    TdgGate,
+    SXGate,
+    SXdgGate,
+)
+from qiskit.circuit.quantumcircuitdata import QuantumCircuitData
+
+INSTRUCTION_TYPE_TO_NAME: Dict[Type[Instruction], str] = {
```

Comment:
I'd rather to have here just the list of gates for which we can't use their default name, for example `SXGate` or even `SWAP`. For the rest, instead of explicitly listing them I would just automatically infer the name from the `Instruction` itself.

Reply:
Please take a look at commit https://github.com/microsoft/quantum-viz.js/pull/44/commits/578e280463b11b59d30b6e1f5aa42a2d7bdbb598

Do you prefer the situation before? I think that we can capitalize all the names except for special cases (swap, sx)